### PR TITLE
terminal: Go syntax highlighting for listings

### LINKED ISSUE #1273: Syntax highlighting
Please answer the following before submitting your issue:

Note: Please include any substantial examples (debug session output,
stacktraces, etc) as linked gists.

1. What version of Delve are you using (`dlv version`)?
```
Delve Debugger
Version: 1.0.0
Build: $Id: c98a142125d0b17bb11ec0513bde346229b5f533 $
```
2. What version of Go are you using? (`go version`)?
`go version go1.10.2 linux/amd64`
3. What operating system and processor architecture are you using?
Ubuntu 18.04 LTS x86_64
4. What did you do?
`l`
5. What did you ~expect~ want to see?
Syntax highlighted code
6. What did you see instead?
Non syntax highlighted code

I now this is difficult but it would be quite useful.

### PR DESCRIPTION
**Pull request by aarzilli**
Fixes #1273

### CODE REVIEW

**Comment by derekparker** [comment]
Seems like the "package" keyword is not being colored correctly:

Screen Shot 2021-01-27 at 7 12 30 AM

**Comment by derekparker** [comment]
Return value types are not being highlighted properly:

Screen Shot 2021-01-28 at 6 51 14 AM

**Comment by aarzilli** [comment]
Builtin types and functions are not highlighted anywhere since they are not keywords or literals. There's a TODO item in colorize.go about it. Doing it correctly 100% of the time would require some amount of typechecking, although I bet nobody knows you can do this:

https://play.golang.org/p/gOaFKs1BQJs

**Comment by derekparker** [comment]
> Builtin types and functions are not highlighted anywhere since they are not keywords or literals. There's a TODO item in colorize.go about it. Doing it correctly 100% of the time would require some amount of typechecking, although I bet nobody knows you can do this:
> 
> https://play.golang.org/p/gOaFKs1BQJs

Diabolical.

**Review by derekparker — APPROVED** [review]
LGTM

**Comment by theycallmejpo** [comment]
Thanks for this! I upgraded to 1.6.0 and my strings are highlighted now. Is there an extra flag / env var I need to set to get the full syntax highlighting or change themes?

File: pkg/terminal/terminal.go [hunk 1]
```diff
@@ -84,30 +85,41 @@ func New(client service.Client, conf *config.Config) *Term {
...
+			if s == "" {
+				return fmt.Sprintf(terminalHighlightEscapeCode, defaultCode)
+			}
+			return s
+		}
+		t.colorEscapes[colorize.KeywordStyle] = conf.SourceListKeywordColor
+		t.colorEscapes[colorize.StringStyle] = wd(conf.SourceListStringColor, ansiBrGreen)
+		t.colorEscapes[colorize.NumberStyle] = conf.SourceListNumberColor
+		t.colorEscapes[colorize.CommentStyle] = wd(conf.SourceListCommentColor, ansiBrMagenta)
+		t.colorEscapes[colorize.ArrowStyle] = wd(conf.SourceListArrowColor, ansiBrYellow)
```
Comment (krobelus) [inline pkg/terminal/terminal.go:111]: On a light-theme terminal, the bright yellow arrow is barely visible.
Additionally, the bright green for strings from line 108 is too bright.
This is on Konsole's "Black on White" theme, which is the default on OpenSUSE. It also looks the same on default xterm.
We can't please everyone, but I think it's possible to find default colors that work for the most common default colorschemes out there (white-on-black, black-on-white and maybe the Ubuntu one with purple background).

**Pull request merged**